libwallet/src/internal/scan.rs

```diff
@@ -76,25 +74,20 @@ where
 
 	let legacy_builder = proof::LegacyProofBuilder::new(keychain);
 	let builder = proof::ProofBuilder::new(keychain);
-	let legacy_version = HeaderVersion(6);
 
 	for output in outputs.iter() {
 		let (commit, proof, is_coinbase, height, mmr_index) = output;
 		// attempt to unwind message from the RP and get a value
 		// will fail if it's not ours
 		let info = {
-			// Before HF+2wk, try legacy rewind first
-			let info_legacy = if valid_header_version(*height, legacy_version) {
-				proof::rewind(keychain.secp(), &legacy_builder, *commit, None, *proof)?
-			} else {
-				None
-			};
```

Comment:
Was this a section that checks if the blockchain was constructed on v2? I mean, what exactly is the "legacy_version"?

Reply:
All blocks have the HeaderVersion in their structure, however the rewind function does not check this version and just does what it needs to do.

Currently the blocks are with HeaderVersion(6), this version is defined in the variable defined inside https://github.com/EpicCash/epic/blob/master/core/ src/consensus.rs, where we have the following piece of code:

```rust
...
/// Mainnet first hard fork height
/// Doubled from the previous hard coded value 1300000
/// We might need to change this later for final release
pub const MAINNET_FIRST_HARD_FORK: u64 = 2600000;

/// Floonet first hard fork height
pub const FLOONET_FIRST_HARD_FORK: u64 = 25800;

/// AutomatedTesting and UserTesting first hard fork height.
pub const TESTING_FIRST_HARD_FORK: u64 = 25800;

/// Get the height of the first epic hard fork
pub fn first_fork_height() -> u64 {
	match global::CHAIN_TYPE.read().clone() {
		global::ChainTypes::Mainnet => MAINNET_FIRST_HARD_FORK,
		global::ChainTypes::Floonet => FLOONET_FIRST_HARD_FORK,
		global::ChainTypes::AutomatedTesting | global::ChainTypes::UserTesting => {
			TESTING_FIRST_HARD_FORK
		}
	}
}

/// Compute possible block version at a given height
pub fn header_version(height: u64) -> HeaderVersion {
	if height < first_fork_height() {
		HeaderVersion(6)
	} else {
		HeaderVersion(7)
	}
}

/// Check whether the block version is valid at a given height, implements
/// 6 months interval scheduled hard forks for the first 2 years.
pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
	version == header_version(height)
}
...
```

We can see in my changes that it still uses legacy_builder, however it is the last resort inside "if".